tools/generator/src/DTO/XcodeScheme.swift

```diff
@@ -20,11 +28,112 @@ struct XcodeScheme: Equatable, Decodable {
     }
 }
 
+extension XcodeScheme {
+    /// Create a new scheme applying any default actions based upon the current scheme.
+    var withDefaults: XcodeScheme {
+        get throws {
+            var buildTargets = [BazelLabel: XcodeScheme.BuildTarget]()
+
+            func enableBuildForValue(
+                _ label: BazelLabel,
+                _ keyPath: WritableKeyPath<XcodeScheme.BuildFor, XcodeScheme.BuildFor.Value>
+            ) throws {
+                do {
+                    try buildTargets[label, default: .init(label: label, buildFor: .init())]
+                        .buildFor[keyPath: keyPath]
+                        .merge(with: .enabled)
+                } catch XcodeScheme.BuildFor.Value.ValueError.incompatibleMerge {
+                    throw PreconditionError(message: """
```

Comment:
Just confirming this is not a user facing error. User facing errors are thrown a different way and use Starlark based verbiage.

Reply:
It could be. Can you send a pointer to a user-facing error that you would like me to mimic?

Reply:
@brentleyjones I should have mentioned you in my last response.

Reply:
The main difference is you need to throw a [UsageError](https://github.com/buildbuddy-io/rules_xcodeproj/blob/e233fd39125707cb6a88123271b6c9c53583dfd5/tools/generator/src/Errors.swift#L4). And the message has to be from the point of view of the BUILD file, mentioning what is wrong and how it can be fixed.

Reply:
I will take a look at some examples in the repo and adjust this one.

Reply:
I updated it to be a `UsageError`. Let me know what you think.